hq_superset/__init__.py

```diff
@@ -22,6 +22,13 @@ def flask_app_mutator(app):
     app.strict_slashes = False
     override_jinja2_template_loader(app)
 
+    # A proxy (maybe) is changing the URL scheme from "https" to "http"
+    # on commcare-analytics-staging.dimagi.com, which breaks the OAuth
+    # 2.0 secure transport check despite transport being over HTTPS. I
+    # hate to do this, but werkzeug.contrib.fixers.ProxyFix didn't fix
+    # it. So I've run out of better options. (Norman 2024-03-13)
+    os.environ['AUTHLIB_INSECURE_TRANSPORT'] = '1'
```

Comment:
Hey @kaapstorm 

Good find on the proxy redirection.

I can make peace with this change to unblock QA/UAT.

It's a positive that this is on analytics server that we have control/access to and not superset.

I was trying to find where in nginx conf, this is getting set and compare that with what we have for HQ though HQ nginx template has a lot happening.

Couple of things I noticed that could be faulty
1. we have two "server" blocks with the same server name in nginx conf. I am not sure if that is expected/correct
2. I don't know what the following block does
```
if (!-f $request_filename) {
      proxy_pass http://superset;
      break;
    }
```

@sravfeyn assuming you have more insight when this was setup.
Can you see something faulty with the current nginx conf or point out what could redirecting urls to http?
I don't know what this block does

Reply:
I took a quick look at the nginx config, and realised it will take me a bit longer to change and test it. I'll merge this PR, and I'll create a follow up ticket to see how we can configure nginx so that we can use werkzeug's "ProxyFix" class (or not have to use it at all) instead of using this environment variable.

Reply:
Okay @kaapstorm, thanks for taking a look.

I am fine with this merge till this stays on staging and is not released to production environment until the secure fix is done.